docs/cvl/cvl2/migration.md

```diff
@@ -28,34 +51,104 @@ Run `python3 CVL1_to_CVL2.0_syntax_update.py --help` for further instructions.
 
 The migration script only handles simple cases, and is not guaranteed to work.
 Some manual work and adjustment may be needed after running the script. The
-script may also make odd mistakes. 
+script may also make odd mistakes.
 
 The script will attempt to make the following changes:
  - replace `sinvoke f(...)` with `f(...)`
  - replace `invoke f(...)` with `f@withrevert(...)`
- - replace `f(...).selector` with `sig:f(...).selector`
+ - replace `f(...)`.selector with `sig:f(...).selector`
  - ensure that rules start with `rule`
- - 
+ - replace `static_assert` with `assert`
+ - replace `static_require` with `require`
+ - add `;` to the end of `pragma`, `import`, `using`, and `use` statements
+ - add a `;` to the end of a methods block entry if it doesn't seem to continue to the next line
+ - add `function` to the beginning of a methods block entry
+ - add `external` to unsummarized methods block entries
+ - change `function f(...)` to `function _.f(...)` for summarized external functions
 
 In particular, as the script only consumes spec files, there are decisions that
 it cannot make, as they are based on the Solidity code. Some of those are
 listed here.
 
-## Method declarations
+## Step 3: Fix type errors
 
-- for `public` functions, the method can be summarized either externally or
-  internally. Annotate visibility of `external` or `internal`, and if you need
-  both, repeat the declaration in the `methods` block.
+This is a good time to try running `certoraRun` on your spec.  The command-line
+interface to `certoraRun` has not changed in CVL 2, so you should try to verify
+your contract the same way you usually would.
 
-## Arithmetic and casts
+If your spec verifies without errors, move on to
+{ref}`cvl2-migration-summaries`!  If `certoraRun` reports errors, you will need
+to fix them manually.  Here are some of the more common errors that you may
+come across:
 
-See <changes> section.
+```{contents}
+:local:
+:depth: 1
+```
 
-```{todo}
-This is incomplete
+This section contains specific advice for these situations; if you come across
+problems that are not covered here, consult the {doc}`changes` or ask!
+
+### Syntax errors introduced by the migration script
+
+The migration script is not perfect, and might miss the occasional semicolon or
+fail to add a keyword when it should.  If you have syntax errors you can't
+understand, consult {ref}`cvl2-superficial-syntax-changes`.
+
+### Type errors in arithmetic and casts
+
+CVL 2 is more careful about converting between different integer types.  See
+{ref}`cvl2-integer-types` in the changes guide for complete details and
+examples.
+
+If you have errors that indicate problems with number types, try the following:
+
+ - Try to change most of your integers to `mathint`.  The only integers that
```

Comment:
maybe it's worth bringing up an example where we have:
```
uint256 x = getUint256();
uint128 y = getUint128();
require x > y;
sendAlong(x,y);
```

This one can be converted safely like this:
```
mathint x = getUint256();
mathint y = getUint128();
require x > y;
sendAlong(assert_uint256(x),assert_uint128(y));
```
and this saves `to_mathint`ing in the `require`.

If however one wishes to pick a non-deterministic `uint256`, it's better to declare it as `mathint` and `require_uint256` as necessary.

Reply:
I think `to_mathint` is preferable in this instance, no?  It's safer and less verbose.  At least, it doesn't seem any worse to me?